static/js/getUserMediaPolyfill.js

```diff
@@ -23,8 +23,33 @@
       };
     } else {
       navigator.mediaDevices.getUserMedia = function() {
+        // A missing `getUserMedia` seemingly can mean one of two things:
+        //
+        // 1) WebRTC is unsupported or disabled on this browser
+        // 2) This is an insecure connection
+        //   * This handling of insecure connections happens only on certain browsers.
+        //     It was observed in Chromium 80 and Firefox 75, but not Firefox 68. I suspect it's the new behavior.
+        //   * In other browsers, it handles insecure connections by throwing `NotAllowedError`.
+        //     We still handle this case in the calling function.
+        //   * See: https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia
+        //     As of this writing, this claims that the browser does *both* of these things for
+        //     insecure connections, which is of course impossible and thus confusing.
+        //
+        // We will attempt to distinguish these two cases by checking for various webrtc-related fields on
+        // `window` (inspired by github.com/muaz-khan/DetectRTC). If none of those fields exist, we assume
+        // that WebRTC is not supported on this browser.
         return new Promise(function(resolve, reject) {
-          reject("getUserMedia is not supported in this browser.");
+          if (!(window.RTCPeerConnection || window.webkitRTCPeerConnection || window.mozRTCPeerConnection || window.RTCIceGatherer)) {
+            var e = new Error("getUserMedia is not supported in this browser.");
+            // I'd use NotSupportedError but I'm afraid they'll change the spec on us again
+            e.name = 'CustomNotSupportedError';
```

Comment:
Best option I could think of at the time, but I'm open to other things. Perhaps just a better name.